Classes/Domain/Index/TcaIndexer/TcaTableService76.php

```diff
@@ -0,0 +1,379 @@
+            $whereClause .= BackendUtility::BEenableFields('pages')
+                . BackendUtility::deleteClause('pages')
+                ;
+        }
+
+        $userDefinedWhere = $this->configuration->getIfExists(
+            'indexing.' . $this->getTableName() . '.additionalWhereClause'
+        );
+        if (is_string($userDefinedWhere)) {
+            $whereClause .= ' AND ' . $userDefinedWhere;
+        }
+        if ($this->isBlacklistedRootLineConfigured()) {
+            $whereClause .= ' AND pages.uid NOT IN ('
+                . implode(',', $this->getBlacklistedRootLine())
+                . ')'
+                . ' AND pages.pid NOT IN ('
+                . implode(',', $this->getBlacklistedRootLine())
+                . ')';
+        }
+
+        $this->logger->debug('Generated where clause.', [$this->tableName, $whereClause]);
+        return $whereClause;
+    }
+
+    public function getFields() : string
+    {
+        $fields = array_merge(
+            ['uid','pid'],
+            array_filter(
+                array_keys($this->tca['columns']),
+                function ($columnName) {
+                    return !$this->isSystemField($columnName)
+                        && !$this->isUserField($columnName)
+                        && !$this->isPassthroughField($columnName)
+                        ;
+                }
+            )
+        );
+
+        foreach ($fields as $key => $field) {
+            $fields[$key] = $this->tableName . '.' . $field;
+        }
+
+        $this->logger->debug('Generated fields.', [$this->tableName, $fields]);
+        return implode(',', $fields);
+    }
+
+
+    /**
+     * Generate SQL for TYPO3 as a system, to make sure only available records
+     * are fetched.
+     */
+    protected function getSystemWhereClause() : string
+    {
+        $whereClause = '1=1'
+            . BackendUtility::BEenableFields($this->tableName)
+            . BackendUtility::deleteClause($this->tableName)
+            . ' AND pages.no_search = 0'
+            ;
+
+        if ($this->tableName !== 'pages') {
+            $whereClause .= BackendUtility::BEenableFields('pages')
+                . BackendUtility::deleteClause('pages')
+                ;
+        }
+
+        return $whereClause;
+    }
+
+    protected function isSystemField(string $columnName) : bool
+    {
+        $systemFields = [
+            // Versioning fields,
+            // https://docs.typo3.org/typo3cms/TCAReference/Reference/Ctrl/Index.html#versioningws
+            't3ver_oid', 't3ver_id', 't3ver_label', 't3ver_wsid',
+            't3ver_state', 't3ver_stage', 't3ver_count', 't3ver_tstamp',
+            't3ver_move_id', 't3ver_swapmode',
+            $this->tca['ctrl']['transOrigDiffSourceField'],
+            $this->tca['ctrl']['cruser_id'],
+            $this->tca['ctrl']['fe_cruser_id'],
+            $this->tca['ctrl']['fe_crgroup_id'],
+            $this->tca['ctrl']['languageField'],
```

Comment:
Wasn't this removed for the 8.7 part? Shouldn't this be removed here too?

https://github.com/Codappix/search_core/pull/132/files

Reply:
Maybe you could move that part to an trait, so that it is not defined in 2 places?

You could do that for other parts as well.

Reply:
Why did you not extend the normal TcaTableService and only change the stuff that was needed?

Reply:
I want to get rid of the old 76 implementations as soon as possible. I'll not add further code that will produce another cleanup.

Instead I've added a test for language handling and removed the line.